x-pack/plugin/eql/src/main/java/org/elasticsearch/xpack/eql/execution/sequence/KeyToSequences.java

```diff
@@ -34,23 +40,42 @@
             this.groups = new SequenceGroup[stages];
         }
 
-        void add(int stage, Sequence sequence) {
+        long add(int stage, Sequence sequence) {
             // create the group on demand
             if (groups[stage] == null) {
                 groups[stage] = new SequenceGroup();
             }
             groups[stage].add(sequence);
+            return groups[stage].ramBytesUsed();
         }
 
-        public void remove(int stage) {
+        public long remove(int stage) {
+            long ramBytesUsed = groups[stage].ramBytesUsed();
             groups[stage] = null;
+            return ramBytesUsed;
         }
 
-        void until(Ordinal ordinal) {
+        long until(Ordinal ordinal) {
             if (until == null) {
                 until = new UntilGroup();
             }
             until.add(ordinal);
+            return until.ramBytesUsed();
+        }
+
+        @Override
+        public long ramBytesUsed() {
```

Comment:
Since the ram used is computed in a batch, there's no point in keeping track per remote/until/add method. It's either incremental or a batch - currently it's a mix.